R/angles.R

```diff
@@ -0,0 +1,101 @@
+calculate.angles.radial <- function(v, tree, spread, total.angle) {
+    root.node.id <- v$id[[1]];
+    node.ids <- c(root.node.id);
+
+    total.angle <- total.angle * spread;
+    angles <- numeric(nrow(tree));
+
+    while (length(node.ids) > 0) {
+        current.node.id <- node.ids[1];
+        node.ids <- node.ids[-1];
+
+        parent.id <- tree$parent[tree$tip == current.node.id];
+
+        if (parent.id == -1) {
+            tree$angle[tree$tip == current.node.id] <- 0;
+            }
+
+        child.ids <- tree$tip[tree$parent == current.node.id & !is.na(tree$parent)];
+        num.children <- length(child.ids);
+
+        if (length(child.ids) > 0) {
```

Comment:
Super-micro optimization - you could use `num.children` in the comparison to avoid paying the (crushing, crippling) cost of `length` twice.

Reply:
Oops! Missed a few more `length` calls. Good catch.